common/src/api/external/mod.rs

```diff
@@ -2123,6 +2145,190 @@ impl std::fmt::Display for Digest {
+    pub id: Uuid,
+    pub address_lot_block_id: Uuid,
+    pub rack_id: Uuid,
+    pub switch_location: String,
+    pub address: IpNet,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct Switchport {
+    pub id: Uuid,
+    pub rack_id: Uuid,
+    pub switch_location: String,
+    pub port_name: String,
+    pub port_settings_id: Option<Uuid>,
+}
+
+#[derive(
+    ObjectIdentity, Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq,
+)]
+pub struct PortSettings {
+    #[serde(flatten)]
+    pub identity: IdentityMetadata,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct PortSettingsInfo {
+    pub settings: PortSettings,
+    pub groups: Vec<PortSettingsGroups>,
+    pub port: PortSettingsPortConfig,
+    pub links: Vec<PortSettingsLinkConfig>,
+    pub link_lldp: Vec<LldpServiceConfig>,
+    pub interfaces: Vec<PortSettingsInterfaceConfig>,
+    pub vlan_interfaces: Vec<VlanInterfaceConfig>,
+    pub routes: Vec<PortSettingsRouteConfig>,
+    pub bgp_peers: Vec<PortSettingsBgpPeerConfig>,
+    pub addresses: Vec<PortSettingsAddressConfig>,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct PortSettingsGroups {
+    pub port_settings_id: Uuid,
+    pub port_settings_group_id: Uuid,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct PortSettingsGroup {
+    #[serde(flatten)]
+    pub identity: IdentityMetadata,
+    pub port_settings_id: Uuid,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum PortGeometry {
+    Qsfp28x1,
+    Qsfp28x2,
+    Sfp28x4,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct PortSettingsPortConfig {
+    pub port_settings_id: Uuid,
+    pub geometry: PortGeometry,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct PortSettingsLinkConfig {
+    pub port_settings_id: Uuid,
+    pub lldp_service_config_id: Uuid,
+    pub link_name: String,
+    pub mtu: u16,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct LldpServiceConfig {
+    pub id: Uuid,
+    pub lldp_config_id: Option<Uuid>,
+    pub enabled: bool,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct LldpConfig {
+    #[serde(flatten)]
+    pub identity: IdentityMetadata,
+    pub chassis_id: String,
+    pub system_name: String,
+    pub system_description: String,
+    pub management_ip: IpNet,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum PortSettingsInterfaceKind {
+    Primary,
+    Vlan,
+    Loopback,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct PortSettingsInterfaceConfig {
+    pub port_settings_id: Uuid,
+    pub id: Uuid,
+    pub interface_name: String,
```

Comment:
It seems like we should use a `Name` here. I.e., is this really any string at all under the customer's control?

Reply:
This type is only returned to the user. It is not used as a parameter that can influence how something gets created.

Reply:
https://github.com/oxidecomputer/omicron/issues/3050